ssl/ssl_lib.c

```diff
@@ -3220,14 +3220,14 @@ char *SSL_get_shared_ciphers(const SSL *s, char *buf, int size)
         return NULL;
 
     for (i = 0; i < sk_SSL_CIPHER_num(clntsk); i++) {
-        int n;
+        unsigned int n;
 
         c = sk_SSL_CIPHER_value(clntsk, i);
         if (sk_SSL_CIPHER_find(srvrsk, c) < 0)
             continue;
 
         n = strlen(c->name);
-        if (n + 1 > size) {
+        if (n >= size) {
```

Comment:
This needs to be more nuianced to avoid compiler complaints.  Something along the lines of: `(n > INT_MAX || (int)n >= size)`

Reply:
I do not see why this should trigger any compiler complaints as is. Can you please elaborate?

Reply:
It's comparing an int with a size_t.  Windows compilers are especially picky about this.

Reply:
> It's comparing an int with a size_t. Windows compilers are especially picky about this.

Actually it is visible in the CI already :grinning: